mongoimport/mongoimport.go

```diff
@@ -556,11 +557,11 @@ func (up *upserter) Insert(doc interface{}) error {
 	var err error
```

Comment:
From what I can tell, this will always be a bson.D because this is called from runInsertionWorker, which reads from a channel of bson.D and passes each document to Insert().

Reply:
Right, but that isn't communicated at the type level or in the documentation. If anything, the type implies the fact that _any_ type should be accepted. Updating this won't have any effect on current behavior for the reason you stated, but it's worth it to remove a bit of code smell (imo).

Reply:
I agree with your comment about code smell. I'm not really sure what we could do to improve the code, though. Because we're calling constructUpsertDocument, which needs to iterate over the document, we need to know what the type is. I think that ideally the flushInserter interface would only accept documents of type bson.D (from the new driver), but this means that some work needs to be done in common to change BufferedBulkInserter to use the new driver's BSON lib as well. Because of this, I'm in favor of leaving this as is until TOOLS-2173 is done and then we can change the interface itself.

Reply:
Ok. For now let's just update the cast so it doesn't have the chance of panicking:
e.g.
```go
document, ok := bson.D(doc)
if !ok {
    // error
}
```